### PR TITLE
[3.8.] No Data Template documentation

### PR DESCRIPTION
**Pull request by ntacheva**
Plus refactoring the current template articles.

### CODE REVIEW

File: _contentTemplates/dropdowns/templates.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#item-template
+The `ItemTemplate` determines how the individual items are rendered in the dropdown element of the component. By default, the text from the particular suggestions is rendered.
```
Comment (dimodi) [inline _contentTemplates/dropdowns/templates.md:2]: "dropdown element" is not very technically precise, so why not just use "dropdown" ?

File: _contentTemplates/dropdowns/templates.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#item-template
```
Comment (dimodi) [inline _contentTemplates/dropdowns/templates.md:1]: We should mention that the template `RenderFragment` exposes a `context` and explain it's the data item object.

File: components/autocomplete/templates.md [hunk 3]
```diff
@@ -12,99 +12,88 @@ position: 20
...
+    <TelerikCheckBox @bind-Value="@IsDataAvailable" OnChange="@OnCheckBoxChangeHandler" />
+    AutoComplete has data
+</p>
+
+<TelerikAutoComplete Data="@AutoCompleteData" @bind-Value="@Role" Placeholder="Write your position">
     <HeaderTemplate>
         <strong>Write your own if you don't see it in the list</strong>
     </HeaderTemplate>
+    <ItemTemplate>
+        Are you a&nbsp;<strong>@context</strong>
```
Comment (dimodi) [inline components/autocomplete/templates.md:56]: Does the `&nbsp;` serve a special purpose here?

**Review by dimodi — CHANGES_REQUESTED** [review]
Explain about the `ItemTemplate` `context`

File: _contentTemplates/dropdowns/templates.md (hunk 1 above)
Comment (ntacheva) [reply _contentTemplates/dropdowns/templates.md]: Fixed.

File: components/autocomplete/templates.md (hunk 3 above)
Comment (ntacheva) [reply components/autocomplete/templates.md]: It is not needed, I removed it from teh examples.

File: _contentTemplates/dropdowns/templates.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+The `ItemTemplate` determines how the individual items are rendered in the dropdown of the component. By default, the text from the particular suggestions is rendered.
+
+The `ItemTemplate` exposes a `context` which represents the data item object. Use it to render the item details.
```
Comment (dimodi) [inline _contentTemplates/dropdowns/templates.md:4]:
```suggestion
The `ItemTemplate` exposes a `context` which represents the data item object. Cast it to the respective model type to access or render the item properties.
```

**Pull request merged**